src/visualization/utilities.jl

```diff
@@ -880,6 +880,41 @@ function unstructured_3d_to_1d_curve(nodes, data, curve, slice, point, nvisnodes
   return calc_arc_length(curve), data_on_curve, mesh_vertices_x
 end
 
+# Convert 3d unstructured data from a general mesh to 2d data at a plane.
+function unstructured_3d_to_2d_plane(nodes, data; elevations=nothing, slice=:yz, point=[0,0,0], nvisnodes=8)
+  # Set dimensions according to 'slice'.
+  if slice === :yz
+    slice_dimension = 1
+    other_dimensions = [2, 3]
+  elseif slice === :xz
+    slice_dimension = 2
+    other_dimensions = [1, 3]
+  elseif slice === :xy
+    slice_dimension = 3
+    other_dimensions = [1, 2]
+  else
+    error("illegal dimension '$slice', supported dimensions are :yz, :xz, and :xy")
+  end
+
+  if elevations != nothing
+    nvisnodes = size(elevations,1)
+    plane = create_plane(nodes[1,:,:,:,:], nodes[2,:,:,:,:], nodes[3,:,:,:,:], point, nvisnodes, slice_dimension, other_dimensions)
+    plane[slice_dimension, :, :] = elevations
+  else
+    plane = create_plane(nodes[1,:,:,:,:], nodes[2,:,:,:,:], nodes[3,:,:,:,:], point, nvisnodes, slice_dimension, other_dimensions)
```

Comment:
Does using `@views` in front of the `plan = ...` statements have an effect on performance/allocations?